src/nodes/conv.rs

```diff
@@ -0,0 +1,384 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License.  You may obtain a copy
+// of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+use fuse;
+use libc;
+use std::fs;
+use std::io;
+use std::os::unix::fs::{FileTypeExt, MetadataExt, PermissionsExt};
+use std::path::Path;
+use std::time::{SystemTime, UNIX_EPOCH};
+use time::Timespec;
+
+/// Fixed point in time to use when we fail to interpret file system supplied timestamps.
+const BAD_TIME: Timespec = Timespec { sec: 0, nsec: 0 };
+
+/// Converts a system time as represented in a fs::Metadata object to a Timespec.
+///
+/// `path` is the file from which the timestamp was originally extracted and `name` represents the
+/// metadata field the timestamp corresponds to; both are used for debugging purposes only.
+///
+/// If the given system time is missing, or if it is invalid, logs a warning and returns a fixed
+/// time.  It is reasonable for these details to be missing because the backing file systems do
+/// not always implement all possible file timestamps.
+fn system_time_to_timespec(path: &Path, name: &str, time: &io::Result<SystemTime>) -> Timespec {
```

Comment:
It's unusual to take a `Result` as a parameter. Perhaps the coercion to an `Option` should be done by the caller? (There's a method for that.)

Reply:
I see `ok()` and `err()`, both which return options for each -- but note that here I'm accessing both values (not just the OK value) because I want to log the error. Given that this is an internal helper function I wouldn't worry too much about this signature. If it has to become public at some point (it might), I'll change it.